### ISSUE TITLE
Géométries complexes

### ISSUE THREAD

**Issue by lgrd** [addok/addok#447]
Bonjour, 

Je voudrais savoir si Addok permet de géolocaliser des rues ou des villes avec leurs géométries. Si j'ai bien compris, on peut le faire avec des points. Mais peut-on charger Addok avec des géométries plus complexes et les avoir dans la réponse ? 

Merci pour votre aide.

**Comment by yohanboniface** [comment]
Désolé, raté cette question :(

Réponse: oui et non ;)

Non, Addok ne sait pas indexer des géométries complexes (le reverse ne fonctionne qu'avec des points, via une table de geohash).

Mais oui, il est possible d'ajouter n'importe quelle propriété aux documents qui sont indexés par addok, lequel va retourner ces propriétés dans la réponse.

Dans le cas d'une propriété géométrique, on voudra peut-être remplacer la Geometry qui est retournée dans le geojson. Ce sera pas fait par addok, mais c'est possible d'ajouter un RESULT_PROCESSOR qui fasse le boulot. Je peux proposer un exemple si ça aide.

Désolé encore d'avoir raté la notif :(

**Comment by lgrd** [comment]
Merci beaucoup pour la réponse ! 
Je comprends l'idée, c'est intéressant :) 

Cela me fait penser à la BAN. Il me semble que la BAN peut stocker plusieurs points pour une seule adresse. On pourrait donc imaginer un processus similaire pour qu'Addok retourne les différents points lorsqu'il renvoit l'adresse. Ou j'ai mal compris ? ^^

**Comment by yohanboniface** [comment]
Oui, absolument.
La solution simple et sans dev supplémentaire, c'est de mettre les infos en plus dans le document indexé et elles se retrouveront dans les properties du geojson.
Pour avoir une geometry geojson adaptée, il faut ajouter un postprocesseur qui prend la main sur la sortie geojson.

**Comment by lgrd** [comment]
Ok ! C'est super ! 

Je suppose que pour le reverse c'est une autre affaire. Il faudrait envisager un développement. 

Merci pour tes réponses !

**Comment by yohanboniface** [comment]
> Je suppose que pour le reverse c'est une autre affaire. Il faudrait envisager un développement.

Oui. Pour des MultiPoint, ça me paraît pas trop compliqué. Pour des Line, Polygon and co, c'est une autre affaire.

**Comment by lgrd** [comment]
Je pensais au multipoint donc ça me va ;) 
Dans mon cas, les géométries "complexes" (line, polygon, ...) sont surtout pour rendre la réponse plus riche que pour faire des traitements lors de la réponse.
